### ISSUE TITLE
API: Adding record comment produces duplicate JSON element with zero TTL

### ISSUE THREAD

**Issue by sshmanko** [PowerDNS/pdns#6204]
- Program: Authoritative
 - Issue type: Bug report

### Short description
Adding comment to existing record via API produces duplicate JSON element with TTL: 0
Clone of #4766

### Environment
 - Operating system:  Centos 6
 - Software version: 4.1.0 (also 4.0.4)
 - Software source: RPM
 - Backend: SQLite

### Steps to reproduce
1. Create example.org zone:
```
curl -X POST --data '{"name":"example.org.", "kind": "Native", "masters": [], "nameservers": ["ns1.example.org.", "ns2.example.org."]}' -v -H 'X-API-Key: changeme' http://127.0.0.1:8081/api/v1/servers/localhost/zones
```

2. Add 2 records without comments (one record has to be with asterisk, second one - CNAME):
```
curl -X PATCH --data '{"rrsets":[{"name":"*.record.example.org.", "type":"A", "ttl":3600, "changetype":"REPLACE", "records":[{"content":"1.2.3.4", "disabled":false}]}]}' -v -H 'X-API-Key: apikey' http://127.0.0.1:8081/api/v1/servers/localhost/zones/example.org
```

```
curl -X PATCH --data '{"rrsets":[{"name":"record.example.org.", "type":"CNAME", "ttl":3600, "changetype":"REPLACE", "records":[{"content":"google.com.", "disabled":false}]}]}' -v -H 'X-API-Key: changeme' http://127.0.0.1:8081/api/v1/servers/localhost/zones/example.org
```

3. Add comment to CNAME record
```
curl -X PATCH --data '{"rrsets":[{"name":"record.example.org.", "type":"CNAME", "ttl":3600, "changetype":"REPLACE", "records":[{"content":"google.com.", "disabled":false}], "comments":[{"content":"asdasdasd", "account":"username", "modified_at": 1000000}]}]}' -v -H 'X-API-Key: changeme' http://127.0.0.1:8081/api/v1/servers/localhost/zones/example.org
```

### Expected behaviour
Comment added to existing record - json zone export contains one CNAME entry

### Actual behaviour
Existing record doesn't have a comment:
```
    {
      "comments": Array[0][
        
      ],
      "name": "record.example.org.",
      "records": Array[1][
        {
          "content": "google.com.",
          "disabled": false
        }
      ],
      "ttl": 3600,
      "type": "CNAME"
    }
```

Invalid additional record created with empty 'records' list and zero TTL:
```
    {
      "comments": Array[1][
        {
          "account": "username",
          "content": "asdasdasd",
          "modified_at": 1000000
        }
      ],
      "name": "record.example.org.",
      "records": Array[0][
        
      ],
      "ttl": 0,
      "type": "CNAME"
    }
```

**Comment by zeha** [comment]
Can confirm on master.

**Comment by robertzajda** [comment]
Hi, it seems that cname record goes to comments table, instead of records. When updating.
Any fixes yet?

**Comment by Habbie** [comment]
Fixed in #4781

**Comment by Habbie** [comment]
Oops. I believe this is fixed in #6780, can you confirm?

**Comment by zeha** [comment]
Appears to be fixed.
